WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -833,36 +833,36 @@ public static void addSite(Activity activity, boolean hasAccessToken, SiteCreati
         }
     }
 
-    @SuppressWarnings("deprecation")
-    private static void showAddSiteDialog(Activity activity, SiteCreationSource source) {
+    private static void showAddSiteDialog(FragmentActivity activity, SiteCreationSource source) {
         DialogFragment dialog = new AddSiteDialog();
         Bundle args = new Bundle();
         args.putString(ARG_SITE_CREATION_SOURCE, source.getLabel());
         dialog.setArguments(args);
-        dialog.show(activity.getFragmentManager(), AddSiteDialog.ADD_SITE_DIALOG_TAG);
+        dialog.show(activity.getSupportFragmentManager(), AddSiteDialog.ADD_SITE_DIALOG_TAG);
     }
 
-    /*
-     * dialog which appears after user taps "Add site" - enables choosing whether to create
-     * a new wp.com blog or add an existing self-hosted one
+    /**
+     * Dialog which appears after user taps "Add site" - enables choosing whether to create
+     * a new wp.com blog or add an existing self-hosted one.
+     *
+     * @apiNote Must pass {@link SitePickerActivity#ARG_SITE_CREATION_SOURCE} in arguments when creating this dialog.
      */
-    @SuppressWarnings("deprecation")
     public static class AddSiteDialog extends DialogFragment {
         static final String ADD_SITE_DIALOG_TAG = "add_site_dialog";
 
         @NonNull
         @Override
-        @SuppressWarnings("deprecation")
         public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+            assert getArguments() != null;
```

Comment:
Thank you for adding this extra check @07jasjeet 👍 
What would you say on replacing it with `requireArguments();`

(thank you @ParaskP7 for the suggestion 🙇)